### PR TITLE
[Windows] Add signature validator for MongoDB

### PR DESCRIPTION
**Pull request by sergei-pyshnoi**
# Description

Adding digital signature validator based on "Get-AuthenticodeSignature" Powershell method. Includes example based on MongoDB installer.

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 [hunk 1]
```diff
@@ -687,3 +704,81 @@ function Get-HashFromGitHubReleaseBody {
...
+            $json = Invoke-RestMethod -Uri "https://api.github.com/repos/${RepoOwner}/${RepoName}/releases?per_page=${SearchInCount}"
+            $tags = $json.Where{ $_.prerelease -eq $IsPrerelease }.tag_name
+            $tag = $tags -match $Version
+            if (-not $tag) {
+                throw "Failed to get a tag name for version $Version."
+            }
+            $releaseUrl = "https://api.github.com/repos/${RepoOwner}/${RepoName}/releases/tag/$tag"
+        }
+    }
+    $body = (Invoke-RestMethod -Uri $releaseUrl).body -replace('`', "") -join "`n"
```
Comment (vpolikarpov-akvelon) [inline images/win/scripts/ImageHelpers/InstallHelpers.ps1:754]: It's not clear why do we need to remove backticks from the request body and why do we join response lines first and splitting them afterwards.

**Review by vpolikarpov-akvelon — CHANGES_REQUESTED** [review]

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 [hunk 2]
```diff
@@ -687,3 +701,23 @@ function Get-HashFromGitHubReleaseBody {
     }
     return $result
 }
+function Test-File-Signature {
```
Comment (vpolikarpov-akvelon) [inline images/win/scripts/ImageHelpers/InstallHelpers.ps1:704]:
```suggestion
function Test-FileSignature {
```
Let's stick to Powershell "Verb-Noun" naming convention

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 [hunk 3]
```diff
@@ -687,3 +701,23 @@ function Get-HashFromGitHubReleaseBody {
...
+ 
+    if ($signature.Status -ne "Valid") {
+        throw "Signature status is not valid"
+    }
+
+    if ($signature.SignerCertificate.Thumbprint.Contains($ExpectedThumbprint) -ne $true) {
+        throw "Signature thumbprint do not match expected"
+    }
+
+    Write-Output "Signature is valid"
```
Comment (vpolikarpov-akvelon) [inline images/win/scripts/ImageHelpers/InstallHelpers.ps1:722]: This message doesn't look informative. Let's either get rid of it or output more information (e.g. name of the file)

**Review by vpolikarpov-akvelon — APPROVED** [review]

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 (hunk 3 above)
Comment (sergei-pyshnoi) [reply images/win/scripts/ImageHelpers/InstallHelpers.ps1]: Add file pathto Write-Output message.

**Review by mikhailkoliada — APPROVED** [review]

**Pull request merged**